lib/util.c

```diff
@@ -648,3 +648,28 @@ xbps_symlink_target(struct xbps_handle *xhp, const char *path, const char *tgt)
 
 	return res;
 }
+
+bool
+xbps_patterns_match(xbps_array_t patterns, const char *path)
+{
+	bool match = false;
+
+	if (patterns == NULL)
+		return false;
+
+	for (unsigned int i = 0; i < xbps_array_count(patterns); i++) {
+		const char *pattern = NULL;
+		bool negate = false;
+		if (!xbps_array_get_cstring_nocopy(patterns, i, &pattern))
+			continue;
+		if (pattern == NULL)
+			continue;
+		if ((negate = *pattern == '!') || *pattern == '\\')
+			pattern++;
+		fprintf(stderr, "pattern=%s path=%s\n", pattern, path);
```

Comment:
This should be rather xbps_dbg_printf if ever. Negate will be useful.

Reply:
Agreed

Reply:
Oh this was from debugging, I can add a debug print if it would be useful.

Reply:
Either way is ok for me

Reply:
But that would require me to add `struct xbps_handle` to the arguments.
When extracting, the callback gets a message about skipped files, I don't really think we need a extra debug print here.

Reply:
get rid of it then